src/main/java/org/opensearch/ad/NodeStateManager.java

```diff
@@ -286,33 +283,58 @@ public void setLastDetectionError(String adID, String error) {
     }
 
     /**
-     * Set last cold start error of a detector
-     * @param adID detector id
-     * @param exception exception, can be null
-     */
-    public void setLastColdStartException(String adID, AnomalyDetectionException exception) {
-        NodeState state = states.computeIfAbsent(adID, id -> new NodeState(id, clock));
-        state.setLastColdStartException(exception);
-    }
-
-    /**
-     * Get last cold start exception of a detector.  The method has side effect.
+     * Get a detector's exception.  The method has side effect.
      * We reset error after calling the method since cold start exception can stop job running.
      * @param adID detector id
-     * @return last cold start exception for the detector
+     * @return the detector's exception
      */
-    public Optional<AnomalyDetectionException> fetchColdStartException(String adID) {
+    public Optional<AnomalyDetectionException> fetchExceptionAndClear(String adID) {
         NodeState state = states.get(adID);
         if (state == null) {
             return Optional.empty();
         }
 
-        Optional<AnomalyDetectionException> exception = state.getLastColdStartException();
-        // since cold start exception can stop job running, we set it to null after using it once.
-        exception.ifPresent(e -> setLastColdStartException(adID, null));
+        Optional<AnomalyDetectionException> exception = state.getException();
+        // since an exception can stop job running, we set it to null after using it once.
```

Comment:
From the old comments "cold start exception can stop job running".  In the PR description "his PR record all exceptions instead of just cold start exceptions in node state.".

Will all exceptions stop job running?

Reply:
only EndRunException can stop job running.  Added comments to clarify.